### DISCUSSION TITLE
PDBs for flux

### DISCUSSION THREAD

**Discussion by saimissioncloud** [fluxcd/flux2#4223] [Q&A]
I don't see flag to add PDB for flux controllers. Is there a flag to install to add PDB for flux controllers?

**Comment by stefanprodan** [comment]
Flux controllers have the `system-cluster-critical` priority class, this should avoid eviction from nodes. You can add PDBs during bootstrap, please see the docs here: https://fluxcd.io/flux/installation/configuration/boostrap-customization/

**Reply by saimissioncloud** [reply]
@stefanprodan we have scaled up flux controllers to 2. All controllers came up fine with 2 replicas except source-controller

source-controller 1st replica looks good but 2nd replica is not coming up

Here is souce-controller logs of the 2nd replica. is this expected?

`{"level":"info","ts":"2023-09-12T15:54:04.838Z","logger":"controller-runtime.metrics","msg":"Metrics server is starting to listen","addr":":8080"}  
{"level":"info","ts":"2023-09-12T15:54:04.838Z","logger":"setup","msg":"caching of Helm index files is disabled"}                                   
{"level":"info","ts":"2023-09-12T15:54:04.839Z","logger":"setup","msg":"starting manager"}                                                          
{"level":"info","ts":"2023-09-12T15:54:04.840Z","msg":"Starting server","kind":"health probe","addr":":9440"}                                       
{"level":"info","ts":"2023-09-12T15:54:04.940Z","msg":"starting server","path":"/metrics","kind":"metrics","addr":":8080"}                          
{"level":"info","ts":"2023-09-12T15:54:04.941Z","logger":"runtime","msg":"attempting to acquire leader lease flux-system/source-controller-leader-el
ection...\n"}    `

**Reply by makkes** [reply]
This is expected, see https://github.com/fluxcd/flux2/issues/1947#issuecomment-943479825

**Reply by saimissioncloud** [reply]
It's fine if it is waiting or in pending state in logs but the pod should be in `Ready` state currently one of the 2 pods is `0/1` state. Can we do something for this. it is not the case with other flux controllers except `source-controller`.

```
NAME                                 READY   STATUS    RESTARTS   AGE
source-controller-68b698c946-744t9   1/1     Running   0          8h
source-controller-68b698c946-nhbt7   0/1     Running   0          6h42m
```

**Reply by hiddeco** [reply]
As the source-controller contains an HTTP server, it should not be in a `Ready` state for as long as it has not won the leadership election as otherwise it would serve traffic (without having artifacts to serve).
